src/api/ans.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { AccountAddress } from "../core";
+import { ANSName, RegisterNameParameters, getOwnerAddress, registerName } from "../internal/ans";
+import { AptosConfig } from "./aptosConfig";
+
+/**
+ * A class to handle all `ANS` operations
+ */
+export class ANS {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  /**
+   * Retrieve the owner address of a domain name or subdomain name.
+   *
+   * ```ts
+   * getOwnerAddress({domainName: "aptos", subdomainName: "test"})
+   * // Will return the owner address of "test.aptos.apt" or undefined
+   * ```
+   *
+   * @param args.domainName - A string of the domain name to retrieve
+   * @param args.subdomainName - A string of the subdomain name to retrieve
+   *
+   * @returns MoveAddressType if the name is owned, undefined otherwise
+   */
+  async getOwnerAddress(args: { name: ANSName }): Promise<AccountAddress | undefined> {
```

Comment:
do we want to return the full AccountAddress object? or just the address (as string)?